lib/utils/conn.go

```diff
@@ -190,7 +190,7 @@ func (w *TrackingWriter) Count() uint64 {
 }
 
 func (w *TrackingWriter) Write(b []byte) (int, error) {
-	n, err := w.w.Write(b)
+	n, err := w.W.Write(b)
```

Comment:
Why export these fields? It allows one to read from the underlying reader/writer directly and break the tracking behavior.

Reply:
The reason for exporting them is that I wanted to access methods on underlying readers. Because of how the interfacing works, there isn't a super clean way to do this that I am aware of.

Reply:
What do you need to access on the underlying readers? If that's truly necessary, I think you need a new type rather than breaking the encapsulation here.

Reply:
In this particular case, it's to allow adding and removing clients from a multiwriter. I could make another type but it'd pretty much be a copy and paste of the trackingwriter code.

Reply:
I've completely revamped how multiplexed io is done which cleans everything up.